reader/components/header/Header.jsx

```diff
@@ -1,41 +1,35 @@
-import React, { useContext } from 'react'
+import React from 'react'
 import { Button } from 'reactstrap'
 import ReactToPrint from 'react-to-print'
 import Icon from '../icons/Icon'
-import GlobalContext from '../../store/configureContext'
 import { downloadPDF } from '../pdf-downloader/PDFDownloader'
 import './Header.scss'
 import Logo from '../logo/Logo'
+import {
+  toggleOutlineSidebar,
+  toggleThumbnailsSidebar,
+} from '../../store/ui/actions'
+import { useGlobalStore } from '../../store'
 
-const Header = () => {
-  const {
-    state: {
-      pdfDocument: { pdfDocumentProxy, pdfPagesLoaded },
-      pdfMetadata,
-      isThumbnailViewVisible,
-      isOutlineViewVisible,
-      printContainerRef,
-    },
-    toggleIsThumbnailViewVisible,
-    toggleIsOutlineViewVisible,
-  } = useContext(GlobalContext)
+const Header = ({ printContainerRef }) => {
+  const [{ metadata, ui, document }, dispatch] = useGlobalStore()
 
   return (
     <div className="header">
       <div className="item d-flex justify-content-start">
         <Button
           title="Show Document Outline"
           color="none"
-          active={isOutlineViewVisible}
-          onClick={toggleIsOutlineViewVisible}
+          active={ui.isOutlineSidebarVisible}
+          onClick={() => dispatch(toggleOutlineSidebar())}
```

Comment:
You are creating a function on each render. For modern machines it's not a big deal but keep it in mind.